### PR TITLE
feat: ✨ add `path_sprout_root()`

### LINKED ISSUE #683: Create and test `path_root_sprout()`
Signature: `path_root_sprout() -> Path:`

Returns: Path object

Context: See the `path_sprout_root()`diagram in the [path functions section](https://sprout.seedcase-project.org/docs/design/implementation/python-functions#path-functions) in the docs. 

Depends on #681 #682

### PR DESCRIPTION
**Pull request by signekb**
## Description

Closes #683 

## Reviewer Focus

This PR needs an in-depth review.

## Checklist

- [X] Added or updated tests
- [X] Tests passed locally
- [X] Linted and formatted code
- [X] Build passed locally
- [X] Updated documentation (= added docstrings)

### CODE REVIEW

File: tests/core/test_path_sprout_root.py
Comment (signekb) [inline tests/core/test_path_sprout_root.py:1]: These tests are basically the same as for `get_root_envvar()`

**Review by lwjohnst86 — APPROVED** [review]
Easy peasy :stuck_out_tongue: nice!

File: sprout/core/path_sprout_root.py
Comment (lwjohnst86) [inline sprout/core/path_sprout_root.py:1]: Oh, forgot to say, uncomment the line in the `__init__.py` file.

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]

**Review by lwjohnst86 — APPROVED** [review]
:tada:

**Pull request merged**